### ISSUE TITLE
Naming of Contrib Implementations

### ISSUE THREAD

**Issue by ArmageddonKnight** [apache/incubator-mxnet#10999]
Dear MXNet Developers,

I have a CUDA implementation of LSTM RNN that is algorithmically the same with `LSTMCell` and `cuDNN LSTM RNN` and I am hoping to contribute it to the MXNet repository (in the `src/operator/contrib` folder, I also prepared its Python interface). Could you please kindly answer my following concerns:

- Would the community accept my contribution?
- Is it acceptable if I name my implementation after the organization that I am working in (e.g., `MyorgLSTMRNN`)? Or should I name it in a more generic way (e.g., `OpenLSTMRNN`)?

Thank you.

**Comment by ankkhedia** [comment]
@sandeep-krishnamurthy Could you please tag this as question

**Comment by szha** [comment]
@ArmageddonKnight Hi Jack. Thanks for offering to contribute. How does your implementation compare to cudnn in terms of supported features, efficiency, etc.?

**Comment by marcoabreu** [comment]
Thanks for your contribution, we definitely would love to look into it and see some benchmarks. About your second question: In general, we try to keep MXNet independent from any company influence and naming, thus we'd prefer a more generic term. Maybe you got a paper or something else that you could use a reference for the naming?

**Comment by ArmageddonKnight** [comment]
@szha @marcoabreu  Thank you so much for your reply. 

@szha My implementation is based on the open-source LSTM RNN implementation https://devblogs.nvidia.com/optimizing-recurrent-neural-networks-cudnn-5/, except that it supports **input dropout** and also **backward pass**. I compared my implementation against `cuDNN 6.0` on the Python interface and it is comparable with `cuDNN`. There are cases in which `cuDNN` is slightly better and worse (both at around `20%`). In my opinion, it is always better for other developers to have a complete baseline implementation to work on and experiment.

@marcoabreu I added my implementation into the Python **word-level language modeling** example for benchmarking. For the naming, if organization dependent naming is not preferred, is `OpenLSTMRNN` acceptable?
